chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -552,19 +555,68 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
         Ok(())
     }
 
-    fn check_transactions(&self, block: &Block) -> Result<(), CheckBlockTransactionsError> {
+    fn check_outputs_timelock(
+        &self,
+        block: &Block,
+        block_height: BlockHeight,
+    ) -> Result<(), CheckBlockTransactionsError> {
+        for tx in block.transactions() {
+            tx.outputs().iter().try_for_each(|output| {
+                match output {
+                    TxOutput::Transfer(_, _)
+                    | TxOutput::Burn(_)
+                    | TxOutput::StakePool(_)
+                    | TxOutput::ProduceBlockFromStake(_, _, _)
+                    | TxOutput::LockThenTransfer(_, _, _) => {}
+                    TxOutput::DecommissionPool(_, _, _, timelock) => match timelock {
+                        OutputTimeLock::ForBlockCount(c) => {
+                            let cs: i64 = (*c).try_into().map_err(|_| {
+                                CheckBlockTransactionsError::InvalidDecommissionMaturityDistanceValue
```

Comment:
Please write the error in the same format as InvalidBlockRewardMaturityDistance. Always convey as much information as possible.